### ISSUE TITLE
Events in records are included in generated Equals

### ISSUE THREAD

**Issue by virzak** [dotnet/roslyn#65170]
**Version Used**: .NET6, .NET 7 rc-2

**Steps to Reproduce**:
https://sharplab.io/#gist:66a3912dd5833af7ac9f64987fb5763e

1. Create a record class
2. Include a public event
3. Create identical records
4. Subscribe to the event
5. Compare the two records
6. They are not equal, due to the event

Declaration
```cs
record MyRecord : INotifyPropertyChanged
{
    public int X { get; set; }
    public int Y { get; set; }
    public event PropertyChangedEventHandler? PropertyChanged;
}
```

Invocation
```cs
MyRecord a = new() { X = 1, Y = 2 }, b = new() { X = 1, Y = 2 };
a.PropertyChanged += (s, e) => { };
var expectTrue = a == b;
```

**Expected Behavior**:
`expectTrue` to be true.

**Actual Behavior**:
`expectTrue` is false.

** Additional information **
- Seems like events were not explicitly discussed in the [records proposal](https://github.com/dotnet/csharplang/blob/f97e2daacca61d82fcd2e05473a07abac92668d2/proposals/csharp-9.0/records.md)
- Previously discussed here including a real life scenario: https://github.com/Fody/PropertyChanged/issues/863#issuecomment-1264768185
- Obviously this can be mitigated by implementing `Equals` by hand or through SG. Nevertheless, the default behaviour seems wrong.

**Comment by jaredpar** [comment]
At this point the behavior is too established to change.

**Comment by dgschrei** [comment]
We just wasted half a day on this gotcha until we finally decided to compary the decompiled sources. 

This is very unexpected behavior and while I can understand that introducing a change to this as a breaking change is undesirable, I would at least ask that this be added in a VERY obvious warning box to the official documentation of how Equals generation works at:

[https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/record](url)

"Having any event in your Record will break your expectation of how Equals works" is severe enough that users should be warned about this and not randomly stumble across it and then need to use dotpeek to even have a chance to find out what is going wrong there.  

In fact adding an analyzer to the Microsoft.CodeAnalysis.Analyzers Nuget Package that complains when events are added to records would also be a good idea.
